plugins/basic/python/tk_maya_basic/plugin_engine.py

```diff
@@ -83,12 +83,14 @@ def bootstrap(sg_user, progress_callback, completed_callback, failed_callback):
 
     logger.debug("Will launch the engine with entity: %s" % entity)
 
-    # Remove the custom logging handler now that the engine will take over logging.
-    sgtk.LogManager().root_logger.removeHandler(plugin_logging_handler)
-
     # Install the bootstrap progress reporting callback.
     toolkit_mgr.progress_callback = progress_callback
 
+    # install a callback that turns off logging just before the
+    # engine starts up its own logging
+    callback = lambda ctx, log_handler=plugin_logging_handler: _pre_engine_start_callback(ctx, log_handler)
```

Comment:
A while back I suggested that the logger should be initialized right after we import sgtk in the Maya startup code. The first few lines should be

```
import sgtk
from . import MayaLoggerHandler
sgtk.LogManager.initialize_base_file_handler("tk-maya.log")
sgtk.LogManager.initialize_custom_handler(MayaLoggerHandler())
mgr = sgtk.bootstrap.ToolkitManager()
...
```

There is not handoff to be done. Logging is set up before we even start bootstrapping.